### ISSUE TITLE
hdop value exposed as integer (194) instead of float (1.94)

### ISSUE THREAD

**Issue by mariocannistra** [mikalhart/TinyGPSPlus#8]
Hi Mikal.
I'm using your wonderful library since September and worked fine.  Thank you!

Today I've found a small issue with "hdop".   I need to log all error info about my GPS (Locosys LS20031) to calculate statistics and then work on sensor fusion with an IMU.
I configured the following custom fields and I'm getting their values without issues:

```
TinyGPSCustom pdop(gps, "GPGSA", 15); // $GPGSA sentence, 15th element
TinyGPSCustom vdop(gps, "GPGSA", 17); // $GPGSA sentence, 17th element
```

Instead, when I try to display  hdop  (already directly managed in your library) i get an integer value.
For example, when a message like this is received:

```
$GPGSA,A,3,04,19,14,22,27,,,,,,,,2.17,1.94,0.98*07
```

I get 2.17 and 0.98 in the custom elements values.
But I get 194 in gps.hdop.value()  .   You can see this behavior using the  KitchenSink.ino  example.

I also tried adding hdop as a custom element:

```
TinyGPSCustom hdop(gps, "GPGSA", 16); // $GPGSA sentence, 16th element
```

but that change breaks the normal processing and i can't get any value updated by the library.

Could you please have a look at the  hdop  processing ?  If it is not a bug and I'm doing something wrong please let me know.

PS: I know I could divide that by 100 but it could also be one of those subtle bugs hiding possible other issues with specific devices or new messages... just in case.

Thank you,
Mario

**Comment by mikalhart** [comment]
Mario, I think the values reported in GPGGA are integral hundredths, while those in GPGSA are floating.  So I think the same values are being reported--just in different formats.  TinyGPS doesn't normally look at GPGSA.

**Comment by mikalhart** [comment]
Hmm... although the example in the SIRF reference manual you linked would suggest otherwise.  I know that in my SIRF-3 EM-406A, the GPGGA HDOP values are integral.

**Comment by Eheran1** [comment]
This is from a MTK3339:
$GPGSA,A,3,10,24,15,18,17,12,,,,,,,1.56,1.28,0.89*05

This is a MT3329:
$GPGSA,A,3,05,12,14,31,29,02,21,26,25,,,,1.48,0.99,1.10*07

In specsheets its always displayed as a decimal number. I have not seen DOP values without the decimal point so far.

**Comment by Eheran1** [comment]
Here is a MT3333 chipset:
$GPGSA,A,3,25,31,20,05,29,26,21,16,,,,,1.14,0.79,0.83*0A

(does anyone know how to use Galileo with the MT3333? Its really weird...

**Comment by mikalhart** [comment]
Fixed! :)  See version 1.0.2.

For backwards compatibility gps.hdop.value() still returns an integer, but the new gps.hdop.hdop() returns the 2-digit decimal value as a float.
